### ISSUE TITLE
USB packet buffer overflow

### ISSUE THREAD

**Issue by elli89** [greatscottgadgets/ubertooth#218]
Today I examined the way USB packets are handled in libubertooth. I learned that the recommended way for using libusb is to poll for received USB packets by periodically calling libusb_handle_events() in a separate event handling thread.

http://libusb.sourceforge.net/api-1.0/group__asyncio.html#eventthread

Currently libubertooth calls libusb_handle_events() in the applications main loop right after processing the previously received packet. That way, if the processing takes too much time, the USB packet will be missed and the user won't be informed about the packet loss.

Based on my latest PR #216 which already offers a thread safe data structure for storing the received USB buffers, I have implemented a libusb event handling thread in libubertooth and tested it with the host tools.

When using ubertooth-rx I get a lot of FIFO overflows while libbtbb calculates the hopping pattern. This is what I would expect because the Ubertooth stick sends 2500 USB packets per second and libbtbb takes about three seconds to calculate the hopping pattern on my computer. On slower and embedded hardware or inside of virtual machines, even more packets might be missed.

Long story short: USB buffer overflows are likely to happen and we should think of a reasonable way to handle them. When we come up with a solution, we could also close the 'uh oh, full_usb_buf not emptied' issues #127 #205 #207 which also address USB buffer overflows.

I think it would be a good idea to inform the user about a packet loss but keep the program running. Lost USB packets are not that bad. In order to prevent future issue reports, the packet loss message should also tell the user that lost packets are not fatal. What do you think? I will also file a PR for my threaded libubertooth so you can have a look at it, test it and tell me what you think of it.

**Comment by mikeryan** [comment]
This should be fixed by #219
